pkg/generators/cluster.go

```diff
@@ -56,9 +57,27 @@ func (g *ClusterGenerator) GenerateApplications(
 	}
 	log.Debug("clusters matching labels", "count", len(clusterSecretList.Items))
 
+	var tmplApplication argov1alpha1.Application
+	tmplApplication.Namespace = appSet.Spec.Template.Namespace
+	tmplApplication.Name = appSet.Spec.Template.Name
+	tmplApplication.Spec = appSet.Spec.Template.Spec
+
+	var resultingApplications []argov1alpha1.Application
+
 	for _, cluster := range clusterSecretList.Items {
+		params := make(map[string]string)
+		params["name"] = cluster.Name
+		params["server"] = string(cluster.Data["server"])
+		for key, value := range cluster.ObjectMeta.Labels {
+			params[fmt.Sprintf("metadata.labels.%s", key)] = value
+		}
 		log.WithField("cluster", cluster.Name).Info("matched cluster secret")
+		tmpApplication, err := utils.RenderTemplateParams(&tmplApplication, params)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
In the list generator, we only log this kind of error
@xianlubird - do we want to stop if there is a problem with ane application or just log it and continue? (I prefer log and continue)